File: Object-C/BlackFireIM/Example/Lib/MSIMSDK-UIKit/ChatDetail/Model/MSLocationMessageCellData.m
//
//  MSLocationMessageCellData.m
//  BlackFireIM
//
//  Created by benny wang on 2021/11/30.
//

#import "MSLocationMessageCellData.h"
#import "MSHeader.h"

@implementation MSLocationMessageCellData

- (MSIMLocationElem *)locationElem
{
    return (MSIMLocationElem *)self.elem;
}

- (CGSize)contentSize
{
   return CGSizeMake(TLocationMessageCell_Width, TLocationMessageCell_Height);
}

- (CGFloat)heightOfWidth:(CGFloat)width
{
    CGFloat height = 0;
    if (self.showName) {
        height += 25;
    }
    CGSize containerSize = [self contentSize];
    height += containerSize.height;
    if (self.direction == MsgDirectionOutgoing) {
        height += 20;
    }
    height += 5 + 5;
    return height;
}

- (NSString *)reuseId
{
    return TLocationMessageCell_ReuseId;
}

@end

File: Swift/MSIM-Swift/Lib/MSIMSDK-UIKit/ChatDetail/Model/MSLocationMessageCellData.swift
//
//  MSLocationMessageCellData.swift
//  MSIM-Swift
//
//  Created by benny wang on 2021/12/9.
//

import UIKit
import MSIMSDK


open class MSLocationMessageCellData: MSMessageCellData {

    public var locationElem: MSIMLocationElem {
        return self.elem! as! MSIMLocationElem
    }
    
    public override func contentSize() -> CGSize {
        return CGSize(width: MSMcros.TLocationMessageCell_Width, height: MSMcros.TLocationMessageCell_Height)
    }
    
    public override func heightOfWidth(width: CGFloat) -> CGFloat {
        var height: CGFloat = 0
        if self.showName {
            height += 25
        }
        let containerSize = self.contentSize()
        height += containerSize.height
        if self.direction == .outGoing {
            height += 20
        }
        height += 5 + 5
        return height
    }
    
    public override var reUseId: String {
        return MSMcros.TLocationMessageCell_ReuseId
    }
}
